core/src/actions/swagger.ts

```diff
@@ -54,76 +49,91 @@ export class Swagger extends Action {
-      api.routes.routes[method]
-        .filter(this.checkSkippedRoutes)
-        .map((route: Route) => {
-          const action = this.getLatestAction(route);
-          if (!action) return;
-
-          const tag = action.name.split(":")[0];
-          const formattedPath = route.path
-            .replace("/v:apiVersion", "") // this is handled by "basePath"
-            .replace(/\/:(\w*)/, "/{$1}");
-
-          // in simpleRouting is enabled, only show the "post" verb for this action, not all 5
-          if (config.servers.web.simpleRouting && method !== "get") {
-            return;
-          }
-
-          swaggerPaths[formattedPath] = swaggerPaths[formattedPath] || {};
-          swaggerPaths[formattedPath][method] = {
-            tags: [tag],
-            summary: action.description,
-            // description: action.description, // this is redundant
-            consumes: ["application/json"],
-            produces: ["application/json"],
-            parameters: Object.keys(action.inputs)
-              .sort()
-              .map((inputName) => {
-                return {
-                  in: route.path.includes(`:${inputName}`) ? "path" : "query",
-                  name: inputName,
-                  type: "string", // not really true, but helps the swagger validator
-                  required:
-                    action.inputs[inputName].required ||
-                    route.path.includes(`:${inputName}`)
-                      ? true
-                      : false,
-                  default:
-                    action.inputs[inputName].default !== null &&
-                    action.inputs[inputName].default !== undefined
-                      ? typeof action.inputs[inputName].default === "object"
-                        ? JSON.stringify(action.inputs[inputName].default)
-                        : typeof action.inputs[inputName].default === "function"
-                        ? action.inputs[inputName].default()
-                        : `${action.inputs[inputName].default}`
-                      : undefined,
-                };
-              }),
-            responses,
-            security: action["permission"]
-              ? [
-                  { GrouparooCSRFTokenAndSessionCookie: [] },
-                  { GrouparooAPIKey: [] },
-                ]
-              : [],
-          };
-
-          if (!tags.includes(tag)) {
-            tags.push(tag);
-          }
-        });
-    });
+    for (const [method, routes] of Object.entries(api.routes.routes)) {
+      routes.map((route) => {
+        const action = this.getLatestAction(route);
+        if (!action) return;
+
+        const tag = action.name.split(":")[0];
+        const formattedPath = route.path
+          .replace("/v:apiVersion", "")
+          .replace(/\/:(\w*)/, "/{$1}")
+          .replace(/\/:(\w*)/, "/{$1}")
+          .replace(/\/:(\w*)/, "/{$1}")
+          .replace(/\/:(\w*)/, "/{$1}")
+          .replace(/\/:(\w*)/, "/{$1}");
```

Comment:
```suggestion
          .replace(/\/:(\w*)/g, "/{$1}")
```

Use global flag to replace all?

Reply:
I *think* this was a problem with the `$1` replacement value - we don't know which index to use to replace things :/   
There's probably a loop we can use here to get them *all* - want to give it a try in the Actionhero repo? 

(edit) made an issue in the Actionhero repo - https://github.com/actionhero/actionhero/issues/2038